### DISCUSSION TITLE
Overwriting Files while OPS Image Creation

### DISCUSSION THREAD

**Discussion by Roopsai507** [nanovms/ops#1620] [Q&A] [answered]
I created a Java image where i have copied the jre and .so files from a ubuntu machine into a CentOS. when I was running the image creation process, I encountered an issue with file overwriting.

Here's the configuration file I used:

```
{
    "Program": "usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java",
    "ProgramPath": "/externallibs",
    "Dirs":["etc","lib","lib64","proc","usr","externallibs"],
    "Args":["com.servers.JettyStart"],
    "RunConfig": {
        "BridgeName": "eth0",
        "Ports": ["8443"]
    },
    "Version" : "1.8.0_191"
}
```

Upon running the image creation command (ops image create -c config.json --imagename=jettytest), I observed the following warning along with some other warnings:
```
warning: overwriting existing file lib64/ld-linux-x86-64.so.2 hostpath old: lib64/ld-linux-x86-64.so.2 new: /lib64/ld-linux-x86-64.so.2
```

This warning indicates that the file ld-linux-x86-64.so.2 was being overwritten during the image creation process where the ubuntu's ld-linux file was replaced by CentOs. However, since the file already exists in the specified lib64 directory, it should not have been overwritten.

This unexpected behavior is preventing the successful starting of the image. Any insights or suggestions on how to resolve this issue?

**Comment by eyberg** [comment, marked as answer]
What's going on is that ops is picking up all the shared libs that are being linked to the Program.

If you are sharing code across instances like this you can either:

a) share the image
or
b) create a package

For more people it probably makes the most sense to create a package instead.
